include/seqan3/io/sequence/sequence_file_format_sam.hpp

```diff
@@ -0,0 +1,352 @@
+#include <seqan3/io/sequence/sequence_file_in_options.hpp>
+#include <seqan3/io/sequence/sequence_file_out_options.hpp>
+#include <seqan3/io/stream/parse_condition.hpp>
+#include <seqan3/range/view/char_to.hpp>
+#include <seqan3/range/view/to_char.hpp>
+#include <seqan3/std/concept/range.hpp>
+#include <seqan3/std/view/subrange.hpp>
+#include <seqan3/std/view/transform.hpp>
+namespace seqan3
+{
+/*!\brief       The sam format.
+ * \implements  sequence_file_format_concept
+ * \ingroup     sequence
+ *
+ * \details
+ *
+ * ### Introduction
+ *
+ * SAM stores sequence alignments to a reference, but is sometimes used to store the seqeuences itself. See the
+ * [article on wikipedia](https://en.wikipedia.org/wiki/SAM_(file_format)) for a an in-depth description of the format.
+ *
+ * ### Fields
+ *
+ * The SAM format provides the fields seqan3::field::SEQ, seqan3::field::ID and seqan3::field::SEQ_QUAL. All fields are required when writing.
+ *
+ * ### Implementation notes
+ *
+ */
+class sequence_file_format_sam
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \brief Rule of five explicitly defaulted.
+     * \{
+     */
+    sequence_file_format_sam() = default;
+    sequence_file_format_sam(sequence_file_format_sam const &) = delete;
+    sequence_file_format_sam & operator=(sequence_file_format_sam const &) = delete;
+    sequence_file_format_sam(sequence_file_format_sam &&) = default;
+    sequence_file_format_sam & operator=(sequence_file_format_sam &&) = default;
+    //!\}
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "sam" },
+    };
+    //!\copydoc sequence_file_in_format_concept::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type,
+              typename seq_qual_type>
+    void read(stream_type                                         & stream,
+              sequence_file_in_options<seq_legal_alph_type> const & options,
+              seq_type                                            & sequence,
+              id_type                                             & id,
+              qual_type                                           & qualities,
+              seq_qual_type                                       & seq_qual)
+    {
+        static_assert(detail::decays_to_ignore_v<seq_type> || detail::decays_to_ignore_v<seq_qual_type>,
+                      "Either the sequence field, or the seq_qual field need to be set to std::ignore.");
+        static_assert(detail::decays_to_ignore_v<qual_type> || detail::decays_to_ignore_v<seq_qual_type>,
+                      "Either the qualities field, or the seq_qual field need to be set to std::ignore.");
```

Comment:
The stuff in sequence io has changed a little so that you don't need to worry about `seq_qual_type` too much anymore. But you need to adapt your code. Have a look at how the fastq code looks now!